### PR TITLE
Invert vScale of compressed textures when invertY is set and invertY …

### PR DESCRIPTION
**Pull request by TrevorDev**
…of ktx textures as they are inverted in the file

http://www.html5gamedevs.com/topic/41063-ktx-textures-loading-missing-files-and-hdr-environmenttexture-are-in-a-boat/?tab=comments#comment-235647

### CODE REVIEW

File: src/Engine/babylon.engine.ts [hunk 1]
```diff
@@ -6182,6 +6182,7 @@ module BABYLON {
 
         private _prepareWebGLTexture(texture: InternalTexture, scene: Nullable<Scene>, width: number, height: number, invertY: boolean, noMipmap: boolean, isCompressed: boolean,
             processFunction: (width: number, height: number, continuationCallback: () => void) => boolean, samplingMode: number = Engine.TEXTURE_TRILINEAR_SAMPLINGMODE): void {
+            invertY = invertY === undefined ? true : invertY;
```
[the remark this answers is not in the document]
Comment (TrevorDev) [reply src/Engine/babylon.engine.ts]: removed, I also broke dds from working so I moved relevant logic to ktx only

**Comment by sebavan** [comment]
Could you also check rebuild on internalTexture ?

I guess you inverse vscale might be missing there ?

**Comment by TrevorDev** [comment]
@sebavan  good call, it looks like this is already handled by createTexture so rebuild worked fine but when testing this using context loss I found another issue which I fixed.

**Comment by sebavan** [comment]
Other quick question why not just inverting invertY instead of relying on texture._invertVScale ???

**Comment by TrevorDev** [comment]
because invertY triggers inversion with UNPACK_FLIP_Y_WEBGL which is not supported by compressed textures so vScale inversion was needed. I originally did this for all compressed textures (like dds) but it broke DDS as the UNPACK_FLIP_Y_WEBGL doesn't work at the start but then works when UploadDDSLevels calls _uploadDataToTextureDirectly.

**Comment by sebavan** [comment]
Agree but could we not swap invertY in the KTX Loader instead of adding a separate flag (just wondering did not check the code enough) ?

**Comment by sebavan** [comment]
Ok got it :-)

**Pull request merged**